src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,9 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()){
```

Comment:
The header comments for this `execute` method must be updated to reflect the change in behavior due to the new `isMutating` method.